src/vectorization/VerletList.cpp

```diff
@@ -0,0 +1,1047 @@
+        {
+          Int3D cellgrid = getSystem()->storage->getInt3DCellGrid();
+          const size_t threshold = std::max(cellgrid[0],std::max(cellgrid[1],cellgrid[2]));
+          size_t cells_left = numRealCells-irow, add_size;
+          if(cells_left <= threshold)
+            // over-estimate with few cells left
+            add_size = max_pairs_per_cell*(numRealCells-irow);
+          else
+            // under-estimate with many cells left
+            add_size = max_pairs_per_cell*(numRealCells-irow) / threshold;
+          size_t old_size = neighborList.nplist.size();
+          size_t new_size = ESPP_FIT_TO_VECTOR_WIDTH(old_size+add_size);
+          neighborList.nplist.resize(new_size);
+          LOG4ESPP_INFO(theLogger,"Resizing neighbor list. num_pairs="<<num_pairs
+            <<" old_size="<<old_size<<" new_size="<<new_size);
+        }
+        ///////////////////////////////////////////////////////////////////////////////////////////
+
+        size_t  cell_id         = cellNborList.cellId(irow);
+        size_t  cell_nnbrs      = cellNborList.numNeighbors(irow);
+        size_t  cell_start      = cell_range[cell_id];
+        size_t  cell_data_end   = cell_range[cell_id+1];
+        size_t  cell_size       = sizes[cell_id];
+        size_t  cell_end        = cell_start + cell_size;
+
+        for(size_t p=cell_start; p < cell_end; p++)
+        {
+          real p_x, p_y, p_z;
+          if(VEC_MODE_AOS)
+          {
+            p_x = position[p].x;
+            p_y = position[p].y;
+            p_z = position[p].z;
+          }
+          else
+          {
+            p_x = pa_p_x[p];
+            p_y = pa_p_y[p];
+            p_z = pa_p_z[p];
+          }
+
+          // self-loop
+          {
+            size_t ncell_id = cell_id;
+
+            size_t  ncell_start     = cell_range[ncell_id];
+            size_t  ncell_data_end  = cell_range[ncell_id+1];
+            size_t  ncell_end       = ncell_start + sizes[ncell_id];
+            int* __restrict npptr =  &(neighborList.nplist[num_pairs]);
+
+            {
+              int ll=0;
+
+              #pragma vector always
+              #pragma vector aligned
+              #pragma ivdep
+              for(size_t np = ncell_start; np<ncell_data_end; np++)
+              {
+                {
+                  int include_np = 1;
+                  real dist_x, dist_y, dist_z;
+                  if(VEC_MODE_AOS)
+                  {
+                    dist_x   = p_x - position[np].x;
+                    dist_y   = p_y - position[np].y;
+                    dist_z   = p_z - position[np].z;
+                  }
+                  else
+                  {
+                    dist_x   = p_x - pa_p_x[np];
+                    dist_y   = p_y - pa_p_y[np];
+                    dist_z   = p_z - pa_p_z[np];
+                  }
+
+                  const real distSqr  = dist_x*dist_x + dist_y*dist_y + dist_z*dist_z;
+
+                  if(p>=np) include_np = 0;
```

Comment:
I suppose these three ifs can be simplified to 
```cpp
if (p < np && distSqr <= cutsq) {
  npptr[ll++] = np;
}

```

Reply:
Simplified if block